decidim-consultations/app/assets/javascripts/decidim/consultations/vote_dialog.js

```diff
@@ -46,4 +46,18 @@ $(document).ready(function () {
     $("#confirm-vote-form-loader,#confirm-vote-form").toggleClass("hide");
     voteConfirmDialog.foundation("close");
   });
+
+  let delegationsButton = $("#delegations-button"),
+      delegationDialog = $("#delegations-modal"),
+      delegationVoteButtons = $(".delegation-vote-button"),
+      delegationField = $("#decidim_consultations_delegation_id");
+
+  delegationsButton.click(function () {
+    delegationDialog.foundation("open");
+  });
+
+  delegationVoteButtons.click(function () {
+    voteDialog.foundation("open");
+    delegationField.val($(this).data("delegation-id"));
+  });
```

Comment:
this needs to be extracted into a JS file in https://github.com/coopdevs/decidim-module-action_delegator/.